wgpu/src/lib.rs

```diff
@@ -1937,15 +1937,27 @@ impl Instance {
     /// - On web: will panic if the `raw_window_handle` does not properly refer to a
     ///   canvas element.
     pub unsafe fn create_surface<
-        W: raw_window_handle::HasRawWindowHandle + raw_window_handle::HasRawDisplayHandle,
+        W: raw_window_handle::HasWindowHandle + raw_window_handle::HasDisplayHandle,
     >(
         &self,
         window: &W,
     ) -> Result<Surface, CreateSurfaceError> {
+        let raw_display_handle = window
+            .display_handle()
+            .map_err(|e| CreateSurfaceError {
+                inner: CreateSurfaceErrorKind::RawHandle(e)
+            })?
+            .as_raw();
+        let raw_window_handle = window
+            .window_handle()
+            .map_err(|e| CreateSurfaceError {
+                inner: CreateSurfaceErrorKind::RawHandle(e)
+            })?
+            .as_raw();
```

Comment:
While we have to wait for https://github.com/rust-windowing/winit/pull/3126 I'd like to start to ask my questions like in this case.

According to the doc of [`HasWindowHandle`] (which deprecates [`HasRawWindowHandle`]):
> Users of this trait will include graphics libraries, like [wgpu](https://crates.io/crates/wgpu) and [glutin](https://crates.io/crates/glutin). These APIs should be generic over a type that implements HasWindowHandle, and should use the [WindowHandle](https://docs.rs/raw-window-handle/0.6.0/raw_window_handle/struct.WindowHandle.html) type to access the window handle. The window handle should be acquired and held while the window is being used, in order to ensure that the window is not deleted while it is in use.

So do you agree here, that this function will return a `CreateSurfaceErrorKind::RawHandle` if it don't receive the handlers?

[`HasWindowHandle`]: https://docs.rs/raw-window-handle/0.6.0/raw_window_handle/trait.HasWindowHandle.html
[`HasRawWindowHandle`]: https://docs.rs/raw-window-handle/0.6.0/raw_window_handle/trait.HasRawWindowHandle.html